lib/wallet/http.js

```diff
@@ -1993,6 +1993,56 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    this.get('/walletbackup', async (req, res) => {
+      const backup = [];
+      const walletIds = await this.wdb.getWallets();
+
+      for (const walletId of walletIds) {
+        if (walletId === 'primary') {
+          continue;
+        }
+
+        const wallet = await this.wdb.get(walletId);
+        const walletBackup = await util.backupWallet(wallet);
+        backup.push(walletBackup);
+      }
+
+      const buffer = util.zipObject(backup);
+      const backupName = `walletBackup_${util.timeStamp()}.json.gzip`;
+      res.setHeader('Content-Disposition', `inline; filename="${backupName}"`);
+      res.buffer(200, buffer);
```

Comment:
We should not send the sensitive wallet details over the wire back to the caller. We should keep it on-disk and let another process deal with storing the file to its eventual secure storage location. Can refer to the code in the full node repo to see how backups currently get uploaded.

But overall comment -- I think we should modify this interface to behave the same as the current wallet backup function, which writes the backup to a file. See the full node repo to understand how wallet backups all fit together or ask in slack

Reply:
fixed on full-node, 

`node.rpc.on('namebase-walletbackup-new',`

handler, shows necessary changes 

**https://github.com/namebasehq/full-node/tree/ridvan-namebase-plugin**